### PR TITLE
Update logging docs

### PR DESCRIPTION
**Pull request by daniellepintz**
## What does this PR do?
Fixes various things in the logging docs.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/extensions/logging.rst [hunk 1]
```diff
@@ -130,19 +130,14 @@ The :meth:`~pytorch_lightning.core.lightning.LightningModule.log` method has a f
 * ``on_step``: Logs the metric at the current step.
 * ``on_epoch``: Automatically accumulates and logs at the end of the epoch.
 * ``prog_bar``: Logs to the progress bar (Default: ``False``).
-* ``logger``: Logs to the logger like ``Tensorboard``, or any other custom logger passed to the
-    :class:`~pytorch_lightning.trainer.trainer.Trainer` (Default: ``True``).
+* ``logger``: Logs to the logger like ``Tensorboard``, or any other custom logger passed to the :class:`~pytorch_lightning.trainer.trainer.Trainer` (Default: ``True``).
```
Comment (daniellepintz) [inline docs/source/extensions/logging.rst:133]: The changes here fix a weird issue where some text appears in a larger font.

Before: https://pytorch-lightning.readthedocs.io/en/latest/extensions/logging.html#automatic-logging

After: https://152279-178626720-gh.circle-artifacts.com/0/html/extensions/logging.html#automatic-logging

File: docs/source/extensions/logging.rst [hunk 2]
```diff
@@ -272,21 +267,6 @@ To change this behaviour, set the ``log_every_n_steps`` :class:`~pytorch_lightni
    trainer = Trainer(log_every_n_steps=k)
 
 
-Log Writing Frequency
```
Comment (daniellepintz) [inline docs/source/extensions/logging.rst:275]: `flush_logs_every_n_steps` is deprecated

Comment (ananthsub) [reply docs/source/extensions/logging.rst:275]: Instead of removing the whole section, could you add that loggers can specify their flush frequency

**Review by ananthsub — COMMENTED** [review]
Comment on flushing section

File: docs/source/extensions/logging.rst [hunk 3]
```diff
@@ -130,19 +130,14 @@ The :meth:`~pytorch_lightning.core.lightning.LightningModule.log` method has a f
...
 * ``on_epoch``: Automatically accumulates and logs at the end of the epoch.
 * ``prog_bar``: Logs to the progress bar (Default: ``False``).
-* ``logger``: Logs to the logger like ``Tensorboard``, or any other custom logger passed to the
-    :class:`~pytorch_lightning.trainer.trainer.Trainer` (Default: ``True``).
+* ``logger``: Logs to the logger like ``Tensorboard``, or any other custom logger passed to the :class:`~pytorch_lightning.trainer.trainer.Trainer` (Default: ``True``).
 * ``reduce_fx``: Reduction function over step values for end of epoch. Uses :meth:`torch.mean` by default.
 * ``enable_graph``: If True, will not auto detach the graph.
-* ``sync_dist``: If True, reduces the metric across devices. Use with care as this may lead to a significant
-    communication overhead.
+* ``sync_dist``: If True, reduces the metric across devices. Use with care as this may lead to a significant communication overhead.
```
Comment (krshrimali) [inline docs/source/extensions/logging.rst:136]: These are some useful fixes! The page looks better now. Thanks!

File: docs/source/extensions/logging.rst [hunk 4]
```diff
@@ -130,19 +130,14 @@ The :meth:`~pytorch_lightning.core.lightning.LightningModule.log` method has a f
...
 * ``sync_dist_group``: The DDP group to sync across.
-* ``add_dataloader_idx``: If True, appends the index of the current dataloader to the name
-    (when using multiple dataloaders). If False, user needs to give unique names for each dataloader to not mix the values.
-* ``batch_size``: Current batch_size used for accumulating logs logged with ``on_epoch=True``.
-    This will be directly inferred from the loaded batch, but for some data structures you might need to explicitly provide it.
-* ``rank_zero_only``: Whether the value will be logged only on rank 0. This will prevent synchronization which
-    would produce a deadlock as not all processes would perform this log call.
+* ``add_dataloader_idx``: If True, appends the index of the current dataloader to the name (when using multiple dataloaders). If False, user needs to give unique names for each dataloader to not mix the values.
+* ``batch_size``: Current batch_size used for accumulating logs logged with ``on_epoch=True``. This will be directly inferred from the loaded batch, but for some data structures you might need to explicitly provide it.
```
Comment (krshrimali) [inline docs/source/extensions/logging.rst:139]:
```suggestion
* ``batch_size``: Current batch size used for accumulating logs logged with ``on_epoch=True``. This will be directly inferred from the loaded batch, but for some data structures, you might need to explicitly provide it.
```

Or:

```suggestion
* ``batch_size``: Current ``batch_size`` used for accumulating logs logged with ``on_epoch=True``. This will be directly inferred from the loaded batch, but for some data structures, you might need to explicitly provide it.
```

File: docs/source/extensions/logging.rst [hunk 5]
```diff
@@ -367,7 +347,7 @@ when using the ``TensorBoardLogger``, all hyperparams will show
 in the `hparams tab <https://pytorch.org/docs/stable/tensorboard.html#torch.utils.tensorboard.writer.SummaryWriter.add_hparams>`_.
 
 .. note::
-    If you want to track a metric in the tensorboard hparams tab, log scalars to the key ``hp_metric``. If tracking multiple metrics, initialize ``TensorBoardLogger`` with ``default_hp_metric=False`` and call ``log_hyperparams`` only once with your metric keys and initial values. Subsequent updates can simply be logged to the metric keys. Refer to the following for examples on how to setup proper hyperparams metrics tracking within :doc:`LightningModule <../common/lightning_module>`.
+    If you want to track a metric in the tensorboard hparams tab, log scalars to the key ``hp_metric``. If tracking multiple metrics, initialize ``TensorBoardLogger`` with ``default_hp_metric=False`` and call ``log_hyperparams`` only once with your metric keys and initial values. Subsequent updates can simply be logged to the metric keys. Refer to the examples below for how to set up proper hyperparams metrics tracking within the :doc:`LightningModule <../common/lightning_module>`.
```
Comment (krshrimali) [inline docs/source/extensions/logging.rst:350]:
```suggestion
    If you want to track a metric in the tensorboard hparams tab, log scalars to the key ``hp_metric``. If tracking multiple metrics, initialize ``TensorBoardLogger`` with ``default_hp_metric=False`` and call ``log_hyperparams`` only once with your metric keys and initial values. Subsequent updates can simply be logged to the metric keys. Refer to the examples below for setting up proper hyperparams metrics tracking within the :doc:`LightningModule <../common/lightning_module>`.
```

**Review by krshrimali — APPROVED** [review]
LGTM: once the review comment by @ananthsub is addressed. Left a few suggestions, thanks for fixing the docs. 🚀

**Review by carmocca — APPROVED** [review]

File: docs/source/extensions/logging.rst (hunk 1 above)
Comment (awaelchli) [reply docs/source/extensions/logging.rst:133]: FYI, this was because the indentation whitespace wasn't the right amount. It has to align properly with the start of the previous line after the bullet point indentation.

**Review by awaelchli — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

File: docs/source/extensions/logging.rst (hunk 2 above)
Comment (daniellepintz) [reply docs/source/extensions/logging.rst:275]: I thought about doing that, but it is really an internal implementation detail of the individual loggers, and I think it makes more sense to go on the docs page for the individual loggers, since it doesn't apply to all loggers. `flush_logs_every_n_steps` is currently **only** on the CSVLogger, and this is stated on the CSVLogger doc: https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.loggers.csv_logs.html#csv-logger

Feel free to disagree though

File: docs/source/extensions/logging.rst (hunk 1 above)
Comment (daniellepintz) [reply docs/source/extensions/logging.rst:133]: I see - it looks like it's fixed now though right?

File: docs/source/extensions/logging.rst (hunk 2 above)
Comment (krshrimali) [reply docs/source/extensions/logging.rst:275]: Hi, @ananthsub - whenever you have the time, can you take a look at the comment here? If it looks good to you, please resolve this and I can enable auto-merge if all looks good to you as well :)

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@daniellepintz there are still two unresolved comments... :rabbit:

File: docs/source/extensions/logging.rst (hunk 2 above)
Comment (daniellepintz) [reply docs/source/extensions/logging.rst:275]: @ananthsub Can you take a look at this please so we can merge it?

Comment (ananthsub) [reply docs/source/extensions/logging.rst:275]: @daniellepintz as a user, I may still land on the logger section for this question. To make life easier for users, I would prefer mentioning that individual logger implementations determine their flushing behavior. This also helps users who set `flush_logs_every_n_steps` before v1.7.

Comment (daniellepintz) [reply docs/source/extensions/logging.rst:275]: sg, updated!

**Comment by daniellepintz** [comment]
Ready to be merged!

**Review by ananthsub — APPROVED** [review]

**Comment by daniellepintz** [comment]
also could someone please rerun tpu tests?

**Comment by akihironitta** [comment]
> also could someone please rerun tpu tests?

@daniellepintz Done!

**Comment by daniellepintz** [comment]
Indeed, thanks @akihironitta!

**Comment by daniellepintz** [comment]
It seems TPU tests are fixed, would someone mind rerunning the TPU tests?
